lmfit/model.py

```diff
@@ -1682,6 +1679,13 @@ def loads(self, s, funcdefs=None, **kws):
             setattr(self, attr, decode4js(modres.get(attr, None)))
 
         self.best_fit = self.model.eval(self.params, **self.userkws)
+        if len(self.userargs) == 2:
```

Comment:
what happens if ```len(self.userargs) != 2``` or does that never happens (sorry, I didn't go through the code completely...

Reply:
Yes, userargs for a Model will be (data, weights) so it should have length == 2 unless someone is really trying to break something.   This could probably be more robust.....

Reply:
that's fine - I just thought that this check here implied that there was another possibility that needed to be treated differently.

Reply:
Honestly, the whole of "dumps()/loads()" for `ModelResult()` seems like a house of cards to me.   Several parts of the result of Model.fit() (say, CompositeModel, MinimizerResult, and Parameters) really seem to be stretching the idea of serialization, or at least my understanding of it.   I'm not confident that the code here is "correct", but it does work and passes the tests we have.